### ISSUE TITLE
Validation set result on the NLQ task using the VSLNet is lower than reported.

### ISSUE THREAD

**Issue by md-mohaiminul** [EGO4D/episodic-memory#9]
Hi,

I tried the VSLNet baseline for the NLQ task and followed the exact steps of https://github.com/EGO4D/episodic-memory/tree/main/NLQ/VSLNet. However, I am getting lower numbers for the validation set than what is reported in the paper. Is there any potential reason for that?

**Comment by satwikkottur** [comment]
Hello @md-mohaiminul ,

Thanks for your interest.
What `max_pos_len` did you use in your experiments? Could you try `128` instead of `512` suggested in the README? I'll update it accordingly.

**Comment by zhongwanjun** [comment]
Hi,

  I also tried to run the VSLNet baseline for the NLQ task and followed the exact steps, and I got similar results 4.34 r@1 IoU 0.3%, which is much lower than the reported results. I also wonder whether there is some potential problem?

**Comment by satwikkottur** [comment]
@zhongwanjun: Could you please check if `max_pos_len=128` has been adjusted.

I'll try to dig deeper into this issue. In the meantime, I'll go ahead and upload the saved model checkpoint used for reporting the results.

**Comment by NNNNAI** [comment]
Hi, thanks for sharing your work.
I followed the instruction of readme,and try to train the valsnet with max_pos_len 128. I check the valset result in epoch 200,step70600, I got 4.52 r@1 IoU 0.3%  and 2.58 r@1 IoU 0.5%. It was much lower than  reported.

**Comment by NNNNAI** [comment]
Here is the detail result of my training model .The command line I used is 
```
python main.py \
    --task nlq_official_v1 \
    --predictor bert \
    --mode train \
    --video_feature_dim 2304 \
    --max_pos_len 128 \
    --epochs 200 \
    --fv official \
    --num_workers 64 \
    --model_dir checkpoints/ \
    --eval_gt_json "data/nlq_val.json"
```

![image](https://user-images.githubusercontent.com/30747212/162703734-fcc26703-ad78-48b9-a920-132a89b7e42e.png)

**Comment by satwikkottur** [comment]
Hello @zhongwanjun @NNNNAI @md-mohaiminul ,

Thanks for your patience. While I look into the issue more deeply, here are few updates/pointers:
1. We've uploaded the checkpoint for VSLNet. You can use the [Ego4D CLI][ego4d_cli] with `--datasets nlq_models` flag to download it and test at your end.

2. As mentioned in the Ego4D paper (page 40), we train the model for 200 epochs, take the checkpoint with best `val` performance and report both `test` and `val` performance. Feel free to use a similar early stopping strategy to pick your best model, as opposed to that at epoch 200.

[ego4d_cli]: https://github.com/facebookresearch/Ego4d/tree/main/ego4d/cli
[ego4d_arxiv]: https://arxiv.org/pdf/2110.07058.pdf

**Comment by NNNNAI** [comment]
> Hello @zhongwanjun @NNNNAI @md-mohaiminul ,
> 
> Thanks for your patience. While I look into the issue more deeply, here are few updates/pointers:
> 
> 1. We've uploaded the checkpoint for VSLNet. You can use the [Ego4D CLI](https://github.com/facebookresearch/Ego4d/tree/main/ego4d/cli) with `--datasets nlq_models` flag to download it and test at your end.
> 2. As mentioned in the [Ego4D paper](https://arxiv.org/pdf/2110.07058.pdf) (page 40), we train the model for 200 epochs, take the checkpoint with best `val` performance and report both `test` and `val` performance. Feel free to use a similar early stopping strategy to pick your best model, as opposed to that at epoch 200.

Thanks for respond. However I check every checkpoint of my trained model, the best performance (R@1 miou0.3) is just 4.6 .Could it be that the version on github is different from the version you trained at that time?

**Comment by Soldelli** [comment]
Hi @satwikkottur thanks for your support. 
I am about to start working on the VLSNet baseline. 
I would like to know if we do have any updates on this issue.
I do not see any change in the codebase since it was raised. 

Can you please let us know what is your timeline on this?
Thanks.
